Assets/Scripts/VFX/PlanetLight.cs

```diff
@@ -0,0 +1,43 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class PlanetLight : MonoBehaviour
+{
+	[Tooltip( "Rate of ratation in degrees per second. Rotation is around the y axis." )]
+	[SerializeField] float _rotationSpeed = 0.1f;
+	[SerializeField] MinMaxF _alphaRange;
+	[SerializeField] MinMaxF _lightIntensityRange;
+
+	private Vector4 _lightDirection;
+	private Renderer _renderer;
+
+	void Awake()
+	{
+		_renderer = GetComponent<Renderer>();
+		_lightDirection = _renderer.material.GetVector("_LightDir");
+	}
+
+	void Update()
+	{
+		// rotate light direction
+		Quaternion rotation = Quaternion.Euler( 0.0f, _rotationSpeed * Time.deltaTime, 0.0f );
+		_lightDirection = rotation * _lightDirection;
+		_renderer.material.SetVector(
+			"_LightDir",
+			new Vector4( _lightDirection.x, _lightDirection.y, _lightDirection.z ) );
```

Comment:
`SetVector()` takes a `Vector4`, I wanted to use a Vector3 for `_lightDirection` to ensure that it would rotate correctly when applying the quaternion, but it occurs to me now that a Vector4 isn't a 4D vector, it's a 3D vector with a homogeneous coordinate, so I'll fix that.